Xen-3.4 Compatibilty (Cowley and before)

Read old version, but dont' do anything.  This will break suspend-resume
and migrate between Cowley- and Boston+.

This patch can be removed in Boston+2.

diff --git a/hw/piix4acpi.c b/hw/piix4acpi.c
index dca0032..55a48b0 100644
--- a/hw/piix4acpi.c
+++ b/hw/piix4acpi.c
@@ -673,6 +673,22 @@ static int pci_devfn_load(QEMUFile* f, void* opaque, int version_id)
     return 0;
 }
 
+/* Xen-3.4 compatibility (Cowley and before):
+ * Read the section, but don't do anything with it. */
+static int pcislots_load_dummy(QEMUFile* f, void* opaque, int version_id)
+{
+    int i;
+    uint8_t dummy;
+    if (version_id != 1)
+        return -EINVAL;
+    for ( i = 0; i < NR_PCI_DEV; i++ ) {
+        qemu_get_8s(f, &dummy);
+    }
+    qemu_get_8s(f, &dummy);
+    qemu_get_8s(f, &dummy);
+    return 0;
+}
+
 static void php_devfn_init(void)
 {
     int i;
@@ -691,6 +707,9 @@ static void php_devfn_init(void)
                           acpi_php_writeb, &php_devfn);
     register_savevm("pci_devfn", 0, 1, pci_devfn_save, pci_devfn_load,
                     &php_devfn);
+    /* Compatibility with Xen 3.4 (Cowley and before) */
+    register_savevm("pcislots", 0, 1, NULL, pcislots_load_dummy,
+                    &php_devfn);
 }
 #endif /* CONFIG_PASSTHROUGH */
 
